### ISSUE TITLE
MD031 should ignore prettier-ignore comment

### ISSUE THREAD

**Issue by plehnen** [DavidAnson/markdownlint#171]
Putting `<!-- prettier-ignore -->` one line above some code block will cause the error **MD031 - Fenced code blocks should be surrounded by blank lines**.

It should ignore html comments when checking for blank lines.

The prettier-ignore is needed, because otherwise prettier would format the code to one line, removing my intended formatting.

**Comment by DavidAnson** [comment]
Can you show an example, please? This sounds kind of like a workaround for a bug in prettier. I may still implement this, but I’d like to see the scenario first. Thanks!

**Comment by plehnen** [comment]
Sure! This is how the part of the file looks like:
<pre>
> Settings > Editor > Live Templates > Vue

- vcomputed

&lt;!-- prettier-ignore --&gt;
```vue
computed: {
  $name$() {
    return this.$data$ $END$;
  },
},
```
</pre>

Removing the prettier-ignore will format the code in a way which is not desired. 
Leaving it there will be warned by markdownlint.
Adding a blank line between the prettier-ignore and the block code will automatically be removed by prettier.

**Comment by DavidAnson** [comment]
Makes sense. Any idea why prettier thinks it can reformat your code fence? It probably should never do that because it’s almost certain to change the way the content renders.

**Comment by plehnen** [comment]
No idea, sorry. I am relatively new to the world of linters ;)
But I feel that a html comment could be placed anywhere and should just be ignored by the linter.
Especially because the comment used as a ignore rule has to be directly attached to the code it belongs to.

**Comment by DavidAnson** [comment]
FYI: I opened https://github.com/prettier/prettier/issues/5971 to track the unexpected reformatting in the `Prettier` project. I'll use this issue to track the above suggestion for `markdownlint`.
